tasks/run.rb

```diff
@@ -99,6 +99,13 @@ def get_result_from_report(last_run_report, run_result, start_time)
         obj.tag = nil if obj.respond_to?(:tag=)
       end
 
+      # check if the run is marked as failed
+      if report.to_ruby['status'] == 'failed'
```

Comment:
nit reports can be large, so maybe only call `to_ruby` once?

Reply:
I updated it to call `to_ruby` only once. But I think it doesn't matter, because both code paths cannot be hit during the same run. when `if report.to_ruby['status'] == 'failed'` is true, the following `return` statement will exit the method, so we don't hit the second `to_ruby`.